src/sentry/static/sentry/app/components/internalStatChart.tsx

```diff
@@ -1,26 +1,33 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 
+import {Client} from 'app/api';
 import MiniBarChart from 'app/components/charts/miniBarChart';
 import LoadingError from 'app/components/loadingError';
 import LoadingIndicator from 'app/components/loadingIndicator';
 import withApi from 'app/utils/withApi';
 
-class InternalStatChart extends React.Component {
-  static propTypes = {
-    api: PropTypes.object.isRequired,
-    since: PropTypes.number.isRequired,
-    resolution: PropTypes.string.isRequired,
-    stat: PropTypes.string.isRequired,
-    label: PropTypes.string,
-    height: PropTypes.number,
-  };
+const defaultProps = {
+  height: 150,
+};
 
-  static defaultProps = {
-    height: 150,
-  };
+type Props = {
+  api: Client;
+  since: number;
+  resolution: string;
+  stat: string;
+  label: string;
+} & Partial<typeof defaultProps>;
```

Comment:
I know that there is some debate over how to properly and @EvanPurkhiser is in favor of limiting use of default props. In that case would it be best to explicitly pass in `height = {150}` for all parent components that render this `InternalStatChart` component?

Reply:
You can pass `height={height || 150}` to `MiniBarChart` as well imo

Reply:
And have `height?: number` in `type Props`

Reply:
I still prefer the defaultProps 😐 let's discuss it in the next tsc 😉

Reply:
following discussions from the tsc i'm gonna choose to make height an optional prop and then default it to 150 if it's null.